lib/internal/child_process.js

```diff
@@ -352,8 +352,8 @@ function onErrorNT(self, err) {
 ChildProcess.prototype.kill = function(sig) {
   var signal;
 
-  if (sig === 0) {
-    signal = 0;
+  if (typeof sig === 'number') {
+    signal = sig;
```

Comment:
Awesome, you just taught me something. I've recently written similar code and used a bunch of `||` conditions.

Modified as requested.